File: 1pset/mario/mario.c
#include <stdio.h>
#include <cs50.h>

void space(int n);
void leftAlign(int temp);
void rightAlign(int h, int temp);

int main(void)
{
    // Get height of pyramids
    int h;
    do
    {
        h = get_int("Height: ");
    }
    while (h < 1 || h > 8);
    
    // Create pyramids
    int temp = 1;
    for (int i = 0; i < h; i++)
    {
        rightAlign(h, temp);
        space(2);
        leftAlign(temp);
        printf("\n");
        temp++; // controls how many # are being printed 
    }
    
}

//Prints n amount of spaces 
void space(int n)
{
    for (int i = 0; i < n ; i++)
    {
        printf(" ");
    }
}

//Aligns # rows to the right
void rightAlign(int h, int temp)
{
    for (int j = 0; j < temp; j++)
    {
        // temp is the number of # to be printed
        // to avoid printing spaces between #
        // print all spaces on the first iteration
        if (j == 0)
        {
            space((h - temp));
        }
        printf("#");
    }
}

//Aligns # rows to the left 
void leftAlign(int temp)
{
    for (int j = 0; j < temp; j++)
    {
        printf("#");
    }
}

File: 6pset/Mario/mario.py
# Prints 2 half-pyramids using hashes (#) 
from cs50 import get_int


def main():
    
    h = 0
    # Get positive integer between 1 and 8, inclusive
    while(True):
        h = get_int("Height: ")
        if h > 0 and h < 9:
            break
        
    s = h
    # Print left and right aligned pyramid 
    for i in range(h):
        s -= 1
        print(" "*s+"#"*(h-s)+" "*2+"#"*(h-s))

        
if __name__ == "__main__":
    main() 